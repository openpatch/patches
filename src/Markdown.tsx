/** @jsx jsx */
import { ReactNode } from "react";
import { jsx, css } from "@emotion/react";
import ReactMarkdown from "react-markdown";
import gfm from "remark-gfm";
import { Blockquote } from "./Blockquote";
import { Code } from "./Code";
import { Table } from "./Table";
import { TableBody } from "./TableBody";
import { TableRow } from "./TableRow";
import { TableCell } from "./TableCell";
import { TableHead } from "./TableHead";

export type MarkdownProps = {
  markdown: string;
};

type CodeBlockProps = {
  value: string;
  language: string;
};

const CodeBlock = ({ value, language }: CodeBlockProps) => (
  <Code language={language}>{value}</Code>
);

export const Markdown = ({ markdown }: MarkdownProps) => {
  return (
    <ReactMarkdown
      plugins={[gfm]}
      renderers={{
        code: CodeBlock,
        blockquote: Blockquote,
        table: Table,
        tableHead: TableHead,
        tableBody: TableBody,
        tableRow: TableRow,
        tableCell: TableCell,
      }}
      source={markdown}
    />
  );
};
