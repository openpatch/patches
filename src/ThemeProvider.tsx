import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
