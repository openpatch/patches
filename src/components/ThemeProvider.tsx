import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
