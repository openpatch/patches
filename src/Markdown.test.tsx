import { render } from "@testing-library/react";
import { Markdown, MarkdownProps } from "./Markdown";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: MarkdownProps) =>
  render(
    <ThemeProvider>
      <Markdown markdown={`# Hi`} {...props} />
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
