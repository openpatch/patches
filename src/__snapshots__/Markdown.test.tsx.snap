// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <h1>
    Hi
  </h1>
</div>
`;
